types/json/binary.go

```diff
@@ -406,6 +406,29 @@ func (bj *BinaryJSON) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// HashValue encode json value, if value type is int64 Will convert to float64
+// Only use in the aggregate executor
+func (bj BinaryJSON) HashValue(buf []byte) []byte {
+	switch bj.TypeCode {
+	case TypeCodeInt64:
+		buf = appendBinaryFloat64(buf, float64(bj.GetInt64()))
```

Comment:
these are all the type codes:

```go
 27 const (
 28     // TypeCodeObject indicates the JSON is an object.
 29     TypeCodeObject TypeCode = 0x01
 30     // TypeCodeArray indicates the JSON is an array.
 31     TypeCodeArray TypeCode = 0x03
 32     // TypeCodeLiteral indicates the JSON is a literal.
 33      const TypeCodeInt64 byte = 9
 34      TypeCodeInt64 indicates the JSON is a signed integer. .
 35     TypeCodeInt64 TypeCode = 0x09
 36     // TypeCodeUint64 indicates the JSON is a unsigned integer.
 37     TypeCodeUint64 TypeCode = 0x0a
 38     // TypeCodeFloat64 indicates the JSON is a double float number.
 39     TypeCodeFloat64 TypeCode = 0x0b
 40     // TypeCodeString indicates the JSON is a string.
 41     TypeCodeString TypeCode = 0x0c
 42 )
```

For simplicity, you can just append the value of `int64` to `buf` instead of converting it to `float64`.

Reply:
ok, thx, I'll try that in the near future.